src/dpdkstat.c

```diff
@@ -0,0 +1,775 @@
+  uint32_t enabled_port_mask;
+  char port_name[RTE_MAX_ETHPORTS][DATA_MAX_NAME_LEN];
+  uint32_t eal_argc;
+  /* Helper info */
+  int   collectd_reinit_shm;
+  pid_t helper_pid;
+  sem_t sema_helper_get_stats;
+  sem_t sema_stats_in_shm;
+  int   helper_pipes[2];
+  enum DPDK_HELPER_STATUS helper_status;
+  enum DPDK_HELPER_ACTION helper_action;
+  /* xstats info */
+  uint32_t num_ports;
+  uint32_t num_xstats;
+  cdtime_t port_read_time[RTE_MAX_ETHPORTS];
+  uint32_t num_stats_in_port[RTE_MAX_ETHPORTS];
+  struct rte_eth_link link_status[RTE_MAX_ETHPORTS];
+  struct rte_eth_xstats xstats;
+  /* rte_eth_xstats from here on until the end of the SHM */
+};
+typedef struct dpdk_config_s dpdk_config_t;
+
+static int g_configured;
+static dpdk_config_t *g_configuration;
+
+static void dpdk_config_init_default(void);
+static int dpdk_config(oconfig_item_t *ci);
+static int dpdk_helper_init_eal(void);
+static int dpdk_helper_run(void);
+static int dpdk_helper_spawn(enum DPDK_HELPER_ACTION action);
+static int dpdk_init(void);
+static int dpdk_read(user_data_t *ud);
+static int dpdk_shm_cleanup(void);
+static int dpdk_shm_init(size_t size);
+
+/* Write the default configuration to the g_configuration instances */
+static void dpdk_config_init_default(void)
+{
+    g_configuration->interval = plugin_get_interval();
+    WARNING("dpdkstat: No time interval was configured, default value %lu ms is set\n",
+             CDTIME_T_TO_MS(g_configuration->interval));
+    /* Default is all ports enabled */
+    g_configuration->enabled_port_mask = ~0;
+    g_configuration->eal_argc = 2;
+    g_configuration->eal_initialized = 0;
+    ssnprintf(g_configuration->coremask, DATA_MAX_NAME_LEN, "%s", "0xf");
+    ssnprintf(g_configuration->memory_channels, DATA_MAX_NAME_LEN, "%s", "1");
+    ssnprintf(g_configuration->process_type, DATA_MAX_NAME_LEN, "%s", "secondary");
+    ssnprintf(g_configuration->file_prefix, DATA_MAX_NAME_LEN, "%s",
+             "/var/run/.rte_config");
+
+    for (int i = 0; i < RTE_MAX_ETHPORTS; i++)
+      g_configuration->port_name[i][0] = 0;
+}
+
+static int dpdk_config(oconfig_item_t *ci)
+{
+  int port_counter = 0;
+
+  /* Initialize a POSIX SHared Memory (SHM) object. */
+  int err = dpdk_shm_init(sizeof(dpdk_config_t));
+  if (err) {
+    DEBUG("dpdkstat: error in shm_init, %s", strerror(errno));
+    return -1;
+  }
+
+  /* Set defaults for config, overwritten by loop if config item exists */
+  dpdk_config_init_default();
+
+  for (int i = 0; i < ci->children_num; i++) {
+    oconfig_item_t *child = ci->children + i;
+
+    if (strcasecmp("Interval", child->key) == 0) {
+      g_configuration->interval =
+            DOUBLE_TO_CDTIME_T (child->values[0].value.number);
+      DEBUG("dpdkstat: Plugin Read Interval %lu milliseconds\n",
+            CDTIME_T_TO_MS(g_configuration->interval));
+    } else if (strcasecmp("Coremask", child->key) == 0) {
+      ssnprintf(g_configuration->coremask, DATA_MAX_NAME_LEN, "%s",
```

Comment:
Please use `sizeof (configuration->coremask)` instead of `DATA_MAX_NAME_LEN`. The same with the subsequent calls to `ssnprintf()` below.